### PR TITLE
Prevents usage of return in main shader functions

### LINKED ISSUE #39863: return in shader results in unexpected behaviour
**Godot version:**
3.2.2

**OS/device including version:**
GLES3
Windows 10 1903
Intel HD 4000

**Issue description:**
`return`in shader results in unexpected behaviour.

bug

**Steps to reproduce:**
1. Add Sprite
2. Add ShaderMaterial to Sprite
3. Add Shader to ShaderMat
4. Write:
```glsl
void fragment() {
	COLOR = vec4(1.0);
	return;
}
```

**Minimal reproduction project:**
[shader_bug.zip](https://github.com/godotengine/godot/files/4839287/shader_bug.zip)

### PR DESCRIPTION
**Pull request by Chaosus**
Fix #39863

### CODE REVIEW

File: servers/rendering/shader_language.cpp [hunk 1]
```diff
@@ -5585,6 +5585,12 @@ Error ShaderLanguage::_parse_block(BlockNode *p_block, const Map<StringName, Bui
 		} else if (tk.type == TK_CF_RETURN) {
 			//check return type
 			BlockNode *b = p_block;
+
+			if (b && b->parent_function && (b->parent_function->name == "vertex" || b->parent_function->name == "fragment" || b->parent_function->name == "light")) {
+				_set_error(vformat("'return' expression is forbidden in '%s' function!", b->parent_function->name));
```
Comment (Anutrix) [inline servers/rendering/shader_language.cpp:5590]: Maybe change it to `"'return' expression is forbidden in processor function '%s'!"`.

Comment (Chaosus) [reply servers/rendering/shader_language.cpp]: Okay

File: servers/rendering/shader_language.cpp [hunk 2]
```diff
@@ -5585,6 +5585,12 @@ Error ShaderLanguage::_parse_block(BlockNode *p_block, const Map<StringName, Bui
 		} else if (tk.type == TK_CF_RETURN) {
 			//check return type
 			BlockNode *b = p_block;
+
+			if (b && b->parent_function && (b->parent_function->name == "vertex" || b->parent_function->name == "fragment" || b->parent_function->name == "light")) {
+				_set_error(vformat("'return' expression is forbidden in processor function '%s'!", b->parent_function->name));
```
Comment (clayjohn) [inline servers/rendering/shader_language.cpp:5590]:
```suggestion
				_set_error(vformat("Using 'return' in '%s' processor function results in undefined behavior!", b->parent_function->name));
```

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**